src/shared/components/gene-search-panel/GeneSearchPanel.tsx

```diff
@@ -0,0 +1,296 @@
+        className={classNames(
+          radioStyles.radioGroupItem,
+          styles.pseudoRadioDisabled
+        )}
+      >
+        <span className={classNames(radioStyles.radio)} />
+        <span className={radioStyles.label}>All Ensembl</span>
+      </div>
+    </div>
+  );
+};
+
+const GeneSearchResults = (props: {
+  speciesList: CommittedItem[];
+  searchResults: SearchResults;
+}) => {
+  const groupedSearchMatches = getGroupedSearchMatches(
+    props.speciesList,
+    props.searchResults.matches
+  );
+
+  return (
+    <table className={styles.resultsTable}>
+      <thead>
+        <tr>
+          <th>
+            Found in
+            <span className={styles.speciesCount}>
+              {groupedSearchMatches.length}
+            </span>
+            species
+          </th>
+          <th>Gene</th>
+        </tr>
+      </thead>
+      <tbody>
+        {groupedSearchMatches.map((rowData) => (
+          <GeneSearchTableRows
+            key={rowData.speciesInfo.genome_id}
+            data={rowData}
+          />
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+const GeneSearchTableRows = (props: { data: SearchMatchesWithSpecies }) => {
+  const { speciesInfo, searchMatches } = props.data;
+
+  const speciesElement = (
+    <div className={styles.speciesCell}>
+      <span>{speciesInfo.common_name ?? speciesInfo.scientific_name}</span>
+      <span className={styles.assemblyName}>{speciesInfo.assembly_name}</span>
+    </div>
+  );
+
+  if (searchMatches.length === 1) {
+    return (
+      <tr>
+        <td>{speciesElement}</td>
+        <td>
+          <GeneMatch match={searchMatches[0]} species={speciesInfo} />
+        </td>
+      </tr>
+    );
+  } else {
+    return (
+      <>
+        {searchMatches.map((match, index) => (
+          <tr key={index}>
+            {index === 0 && (
+              <td rowSpan={searchMatches.length}>{speciesElement}</td>
+            )}
+            <td>
+              <GeneMatch match={match} species={speciesInfo} />
+            </td>
+          </tr>
+        ))}
+      </>
+    );
+  }
+};
+
+const GeneMatch = (props: { match: SearchMatch; species: CommittedItem }) => {
+  const {
+    match: { unversioned_stable_id },
+    species: { genome_tag, genome_id }
+  } = props;
+  const [shouldShowTooltip, setShouldShowTooltip] = useState(false);
+  const anchorRef = useRef<HTMLSpanElement>(null);
+
+  const genomeIdForUrl = genome_tag || genome_id;
+
+  const handleClick = () => {
+    setShouldShowTooltip(!shouldShowTooltip);
+  };
+
+  const urlForGenomeBrowser = urlFor.browser({
+    genomeId: genomeIdForUrl,
+    focus: buildFocusIdForUrl({ type: 'gene', objectId: unversioned_stable_id })
+  });
+
+  const urlForEntityViewer = urlFor.entityViewer({
+    genomeId: genomeIdForUrl,
+    entityId: buildFocusIdForUrl({
+      type: 'gene',
+      objectId: unversioned_stable_id
+    })
+  });
+
+  // NOTE: If this panel opens in genome browser, we should add `replaceState: true`
```

Comment:
What is the use of this?

Reply:
Whether a new history item gets pushed to the browser history stack, or whether it replaces the topmost history item on the stack:

![image](https://user-images.githubusercontent.com/6834224/234647891-5c87f1eb-17d5-432d-aa53-20a6b9ba8dd4.png)

What it means in practical terms is that it determines what happens when you press on web browser's Back button. If you've added the new location via push, you go to location 2 on the diagram above; if via replace, then to location 1.

On the genome browser page, in-app search is using the `replace` mechanism. Maybe we don't need it (we don't use replace when we switch from gene to gene via zmenu or via previously viewed links)